intro-dplyr.Rmd

```diff
@@ -0,0 +1,343 @@
+table4a
+```
+
+- Is `table4a` tidy?
+- How would you like to organise data in `table4`?
+    - Would it be helpful to have columns `country`, `year`, and `cases` with counts?
+    - Gather columns `1999` and `2000` into row values of a new column `year` and cases counts into values of a new column `cases`
+
+```{r}
+table4a_tidy <- table4a %>%
+  gather(key = "year", value = "cases", `1999`:`2000`)
+table4a_tidy
+```
+
+### Exercise: Convert your data into a long form
+
+Time: 5 min
+
+```{r}
+# explore structure of `table4b`
+table4b
+
+# use gather() to make it tidy
+table4b_tidy <- table4b %>%
+  gather(key = "year", value = "populaton", `1999`, `2000`)
+
+table4b_tidy
+```
+
+## Converting to wide form with `spread()`
+
+Explore the structure of the `table2`
+
+```{r}
+table2
+```
+
+- Is `table2` tidy?
+    - Depends on what we plan to use it for.
+    - Later we would like to calculate the rate of the disease in each country per year: `rate = cases / 100,000 * population`.
+
+- What data structure is preferred?
+    - Each country has a row that contains all information needed to proceed with the calculation: year, number of cases, and the population size.
+    - Spread the column `type` into two columns `cases` and `population`.
+    
+```{r}
+table2 %>%
+    spread(key = type, value = count)
+```
+
+## Case study for using tidyr
+
+The `tidyr::who` dataset contains tuberculosis (TB) cases broken down by year, country, age, gender, and diagnosis method.
+
+Explore `who` data: are they tidy?
+
+```{r}
+glimpse(who)
+```
+
+`new_sp_m014` to `newrel_f65` are likely values, not variables. We don't know what those values represent yet, so we'll give them the generic name "key". We know the cells represent the count of cases, so we'll use the variable cases. Let's reshape these data:
+
+```{r}
+who1 <- who %>% 
+  gather(new_sp_m014:newrel_f65, key = "key", value = "cases", na.rm = TRUE)
+who1
+```
+
+The first three letters of each column denote whether the column contains new or old cases of TB. In this dataset, each column contains new cases.
+
+The next two letters describe the type of TB:
+
+- rel = relapse
+- ep = extrapulmonary TB
+- sn = smear negative
+- sp = smear positive
+The sixth letter gives the sex of TB patients. The dataset groups cases by males (m) and females (f).
+
+The remaining numbers gives the age group. The dataset groups cases into seven age groups:
+
+- 014 = 0 - 14 years old
+- 1524 = 15 - 24 years old
+- 2534 = 25 - 34 years old
+- 3544 = 35 - 44 years old
+- 4554 = 45 - 54 years old
+- 5564 = 55 - 64 years old
+- 65 = 65 or older
+
+Fixing inconsistent column names (`new_sp_` but `newrel_`)
+
+```{r}
+who2 <- who1 %>% 
+  mutate(key = gsub(pattern = "newrel", replacement = "new_rel", x = key))
+who2
+```
+
+Fixing an issue of having several values (e.g., description of the case, type of TB, sex, and age) are coded as one value:
+
+```{r}
+who3 <- who2 %>% 
+  separate(key, c("new", "type", "sexage"), sep = "_")
```

Comment:
Not sure I would show this function. You already have a lot of functions to go over.